cpp/src/io/parquet/page_enc.cu

```diff
@@ -239,6 +247,62 @@ struct BitwiseOr {
   }
 };
 
+// I is the column type from the input table
+template <typename I>
+struct delta_enc {
+  using output_type = std::conditional_t<std::is_signed_v<I>, zigzag128_t, uleb128_t>;
+
+  page_enc_state_s<0>* const s;
+  uint32_t const valid_count;
+  uint64_t* const buffer;
+  void* const temp_space;
+
+  __device__ thrust::pair<bool, size_type> calc_idx_and_validity(uint32_t cur_val_idx)
+  {
+    size_type const val_idx_in_block    = cur_val_idx + threadIdx.x;
+    size_type const val_idx_in_leaf_col = s->page_start_val + val_idx_in_block;
+
+    uint32_t const is_valid = (val_idx_in_leaf_col < s->col.leaf_column->size() &&
```

Comment:
why not bool?

Reply:
ah, just wait for it :rofl: I just changed it with the name :D